implementations/rust/ockam/ockam_key_exchange_x3dh/src/responder.rs

```diff
@@ -100,10 +100,11 @@ impl<V: X3dhVault> KeyExchanger for Responder<V> {
                     .sign(identity_secret_key, signed_prekey_pub.as_ref())?;
                 let identity_key = self.vault.secret_public_key_get(identity_secret_key)?;
                 let one_time_prekey_pub = self.vault.secret_public_key_get(one_time_prekey)?;
+                let signature_array = *array_ref![signature, 0, 64];
```

Comment:
Let's check the length of signature, we don't want to panic in case signature of invalid size was passed

Reply:
Okey. Now I have added checking signature length in those places where I copy inner bytes of a signature.
So it should be solved now.